syntax = "proto3";
package chasky;

import "chasky/core/framework/types.proto";

message AttrDef {
  // A description name for this argument.
  string name = 1;

  string type = 2;

  AttrValue value = 3;

  // Human readable description for this attribute
  string doc = 4;
};

message AttrValue {
  oneof value {
    // string type
    bytes string_val = 2;
    int64 int64_val = 3;
    float float_val = 4;
    bool bool_val = 5;
    DataType dtype_val = 6;
    ListValue  list_val = 1;
  }

  message ListValue {
    repeated bytes string_val = 2;
    repeated int64 int64_val = 3 [packed = true];
    repeated float float_val = 4 [packed = true];
    repeated bool bool_val = 5 [packed = true];
    repeated DataType dtype_val = 6 [packed = true];
  }

}